### PR TITLE
US581282: Health Check permissions for all K8s namespaces

### PR DESCRIPTION
**Pull request by jonny-mccoubrey**
https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=581282

### CODE REVIEW

File: autoscale-scaler-kubernetes/src/main/java/com/github/autoscaler/scaler/kubernetes/K8sServiceScaler.java [hunk 1]
```diff
@@ -133,7 +133,7 @@ public InstanceInfo getInstanceInfo(final String resourceId) throws ScalerExcept
     @Override
     public HealthResult healthCheck()
     {
-        return K8sHealthCheck.healthCheck();
+        return K8sHealthCheck.healthCheck(this.config);
```
Comment (dermot-hardy) [inline autoscale-scaler-kubernetes/src/main/java/com/github/autoscaler/scaler/kubernetes/K8sServiceScaler.java]: Just `config` is fine.  The only place I would use `this.` is in the constructor, or in another method if there was some ambiguity.

```suggestion
        return K8sHealthCheck.healthCheck(config);
```

File: autoscale-scaler-kubernetes/src/main/java/com/github/autoscaler/scaler/kubernetes/K8sHealthCheck.java [hunk 2]
```diff
@@ -28,20 +29,22 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.List;
```
Comment (dermot-hardy) [inline autoscale-scaler-kubernetes/src/main/java/com/github/autoscaler/scaler/kubernetes/K8sHealthCheck.java:32]: You're adding an import that you don't use.

File: autoscale-scaler-kubernetes/src/main/java/com/github/autoscaler/scaler/kubernetes/K8sHealthCheck.java [hunk 3]
```diff
@@ -28,20 +29,22 @@
...
+
 final class K8sHealthCheck
 {
-    private static final Logger LOG = LoggerFactory.getLogger(K8sHealthCheck.class);
+    private static final Logger LOGGER = LoggerFactory.getLogger(K8sHealthCheck.class);
```
Comment (dermot-hardy) [inline autoscale-scaler-kubernetes/src/main/java/com/github/autoscaler/scaler/kubernetes/K8sHealthCheck.java]: Renaming the logger is an odd thing to be doing in this PR which is meant to be dealing with the hard-coded namespaces issue.

File: autoscale-scaler-kubernetes/src/main/java/com/github/autoscaler/scaler/kubernetes/K8sHealthCheck.java [hunk 4]
```diff
@@ -28,20 +29,22 @@
...
 
     private K8sHealthCheck()
     {
 
     }
 
-    public static HealthResult healthCheck()
+    public static HealthResult healthCheck(K8sAutoscaleConfiguration config)
```
Comment (dermot-hardy) [inline autoscale-scaler-kubernetes/src/main/java/com/github/autoscaler/scaler/kubernetes/K8sHealthCheck.java]: Parameters should be `final`:

```suggestion
    public static HealthResult healthCheck(final K8sAutoscaleConfiguration config)
```

File: autoscale-scaler-kubernetes/src/main/java/com/github/autoscaler/scaler/kubernetes/K8sHealthCheck.java [hunk 5]
```diff
@@ -53,42 +56,44 @@ private static HealthResult checkConnection()
...
             return HealthResult.RESULT_HEALTHY;
         } catch (KubectlException e) {
-            LOG.warn("Connection failure to kubernetes", e);
+            LOGGER.warn("Connection failure to kubernetes", e);
             return new HealthResult(HealthStatus.UNHEALTHY, "Cannot connect to Kubernetes");
         }
     }
 
-    private static HealthResult checkPermissions()
+    private static HealthResult checkPermissions(K8sAutoscaleConfiguration config)
```
Comment (dermot-hardy) [inline autoscale-scaler-kubernetes/src/main/java/com/github/autoscaler/scaler/kubernetes/K8sHealthCheck.java]: `final`

**Review by dermot-hardy — APPROVED** [review]

**Comment by jonny-mccoubrey** [comment]
@dermot-hardy I have added in a check to ensure an empty list of namespaces is not passed into health check - and added unit test

**Review by dermot-hardy — APPROVED** [review]

**Pull request merged**